src/Compilers/Core/Portable/Syntax/SyntaxNode.cs

```diff
@@ -1251,11 +1251,11 @@ internal SyntaxNode GetNodeWithoutAnnotations(IEnumerable<SyntaxAnnotation> anno
         /// modification, even if the type of a node changes.
         /// </para>
         /// </remarks>
-        public T? CopyAnnotationsTo<T>(T? node) where T : SyntaxNode
+        public T CopyAnnotationsTo<T>(T node) where T : SyntaxNode?
         {
             if (node == null)
             {
-                return null;
+                return node;
```

Comment:
This is a workaround for the fact that the `node == null` test doesn't/can't hint to the analysis that `T` is nullable. It would also be acceptable to use `null!` here.

Reply:
`T` is not necessarily nullable just because `node` is. It could have been passed with a `!` at the call site, for example.